src/storage/storages.rs

```diff
@@ -332,3 +334,191 @@ impl<'a, C, T: UnprotectedStorage<C>> Join for &'a mut FlaggedStorage<C, T> {
     }
 }
 
+/// Wrapper storage that stores modifications to components with a bitset along with
+/// comparing equality to a cached version.
+///
+/// `ChangedStorage::maintain()` should be called after every modification of the storage
+/// to *maintain* the cache and flagging correctly. 
+///
+///# Example Usage:
+/// 
+/// ```rust
+/// extern crate specs;
+/// use specs::prelude::*;
+/// 
+/// #[derive(PartialEq, Clone)]
+/// pub struct Comp(u32);
+/// impl Component for Comp {
+///     // `ChangedStorage` acts as a wrapper around another storage.
+///     // You can put any store inside of here (e.g. HashMapStorage, VecStorage, etc.)
+///     type Storage = ChangedStorage<Comp, VecStorage<Comp>>;
+/// }
+/// 
+/// pub struct CompSystem;
+/// impl<'a> System<'a> for CompSystem {
+///     type SystemData = WriteStorage<'a, Comp>;
+///     fn run(&mut self, mut comps: WriteStorage<'a, Comp>) {
+///         // Iterates over all components like normal.
+///         for comp in (&comps).join() {
+///             // ...
+///         }
+/// 
+///         // To iterate over the flagged/modified components:
+///         for flagged_comp in (&comps).open().1.join() {
+///             // ...
+///         }
+/// 
+///         // Modify the components in some way.
+///         for comp in (&mut comps).join() {
+///             // ...
+///         }
+///         // `maintain` checks for equality between the cache and the actual components 
+///         // and filters out the components that did not change from the flagged ones.
+///         (&mut comps).open().1.maintain();
+///
+///         // Clears the changed storage every frame with this system.
+///         (&mut comps).open().1.clear_flags();
+///     }
+/// }
+///# fn main() { }
+/// ```
+pub struct ChangedStorage<C, U> {
+    inner: FlaggedStorage<C, U>,
+    cache: U,
+    next: Option<Index>,
+}
+
+impl<C, U> ChangedStorage<C, U>
+    where C: PartialEq + Clone,
+          U: UnprotectedStorage<C>,
+{
+    /// Maintains the flagging and cache of the `ChangedStorage`.
+    /// 
+    /// Checks for equality between the cache and the actual components data
+    /// to filter out the components that did not change, leaving only the modified ones
+    /// in the flagged bitset.
+    ///
+    /// Should be called whenever you care about flagged components or alternatively
+    /// whenever the component storage is modified.
```

Comment:
This is very unclear; unless you know the implementation details, this line won't help users. Could you maybe include a usage example?